### PR TITLE
Space Travelers' Hub

### PR DESCRIPTION
**Pull request by Maha-Magdy**
Hello 😊,

In this project, my partner and I have done the following:-
- Config & basic setup.
- Fetch data and update the Redux store.
- Render UI:lists.
- Write actions and reducers for booking rockets and joining missions.
- Write actions and reducers for canceling rockets and leaving missions.
- Render UI: conditional components rendering.
- Render UI: My Profile section.

Thanks, and looking forward to the feedback.

### CODE REVIEW

File: src/components/Profile.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const { missions } = props;
+  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);
+  const reservedMissions = missions.filter((mission) => mission.reserved === true);
+  return (
+    <div className="container-fluid profile">
+      <div className="row align-items-start">
+        <div className="col">
+          <h5>My Missions</h5>
+          {reservedMissions.length === 0 && (
+            <p>There&apos;s no reserved missions</p>
```
Comment (e71az) [inline src/components/Profile.js:16]: - Great job handling no rockets/missions found, it was an optional requirement and you added it!

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️
_Check the comments under the review_
<hr>

Hello @Maha-Magdy @herokudev, I am Elias, your reviewer for this PR

Well done so far with your project!

Your project was almost perfect, congratulations! It is missing some really small details, keep it going!

I will mention to you a small summary of things that need to be changed in order to keep moving:

- [x] Please fix this console warning that is appearing when running your project to comply with Microverse general requirements.
 
image

- [x] Please add an npm install instruction in your readme to let the users know how to correctly run your project.
- [x] Please add unit tests to your application with React testing library and Jest for unit tests to comply with the project requirements.
- [x] Please check the comments below in this PR for extra requirements.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

1. The `My Profile` section can be enhanced by adding the "Cancel reservation" and "Leave Mission" buttons to the lists here. Clicking them should dispatch the actions you have already used in the main Rockets and Missions sections.
2. The `My Profile` section can be enhanced by adding the "Read more" button for each mission and rocket. Upon click, it should open a corresponding Wikipedia page in a new tab. NOTE - you need to get that extra Wikipedia URL from the API's payload.

Please correct those issues and send another review request, good luck, and well done so far! 

Code review by Elias Castañeda

_(Feel free to leave any questions or comments in the PR thread if something is not 100% clear)_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


As described in the [Code reviews limit Policy](https://microverse.zendesk.com/hc/en-us/restricted?return_to=https%3A%2F%2Fmicroverse.zendesk.com%2Fhc%2Fen-us%2Farticles%2F1500004088561) you have a limited number of code reviews per project (you can check the exact number in your Dashboard). If you think that the code review was not fair, you can [request a second opinion using this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: public/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
```
Comment (addod19) [inline public/index.html:1]: - [ ] It would be better to add a unit testing for the application as mentioned in the previous comments

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @Maha-Magdy and @herokudev, 

Great work on making the changes requested by a previous reviewer 👏 Fix console errors :+1: Add npm instructions :100: 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BrianSammit — APPROVED** [review]
# STATUS: APROVED :green_circle: :clap: 

Hello @Maha-Magdy  and @herokudev, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

Well-Done-Applause

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**